api/src/main/java/org/openmrs/ServiceOrder.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+/**
+ * Provides properties for several order types like TestOrder , ReferralOrder
+ * and others to accommodate referrals depending on the openmrs implementation
+ * use case
+ * 
+ * @since 2.5.0
+ */
+public abstract class ServiceOrder extends Order {
+
+	public enum Laterality {
+		LEFT,
+		RIGHT,
+		BILATERAL
+	}
+
+	public static final long serialVersionUID = 1L;
+
+	public Concept specimenSource;
+
+	private Laterality laterality;
+
+	private String clinicalHistory;
+
+	private OrderFrequency frequency;
+
+	private Integer numberOfRepeats;
+	
+	private Location location;
+
+	/**
+	 * Default Constructor
+	 */
+	public ServiceOrder() {}
+
+	/**
+	 * @see org.openmrs.Order#copy()
+	 * <strong>Should</strong> copy all sub-class order fields
+	 */
+	@Override
+	public Order copy() {
+		return copyHelper(new Order());
+	}
+
+	protected abstract Order copyHelper(Order target);
```

Comment:
Why did you leave this one empty?

Reply:
If I turn```
 /**
	 * @see Order#cloneForRevisionHelper(Order)
	 */
	protected TestOrder cloneForRevisionHelper(TestOrder target) {
		super.cloneForRevisionHelper(target);
		target.setSpecimenSource(getSpecimenSource());
		target.setLaterality(getLaterality());
		target.setClinicalHistory(getClinicalHistory());
		target.setFrequency(getFrequency());
		target.setNumberOfRepeats(getNumberOfRepeats());
		return target;
	}
```
to this

```
	protected Order cloneForRevisionHelper(Order target) {
		super.cloneForRevisionHelper(target);
		target.setSpecimenSource(getSpecimenSource());
		target.setLaterality(getLaterality());
		target.setClinicalHistory(getClinicalHistory());
		target.setFrequency(getFrequency());
		target.setNumberOfRepeats(getNumberOfRepeats());
		return target;
	}
```

We cannot get the properties of `TestOrder` , as it's intended to sub-class `ServiceOrder`.

this works , but i wonder wether we can pass **ServiceOrder** as a param


	```
protected Order cloneForRevisionHelper(ServiceOrder target) {
		super.cloneForRevisionHelper(target);
		target.setSpecimenSource(getSpecimenSource());
		target.setLaterality(getLaterality());
		target.setClinicalHistory(getClinicalHistory());
		target.setFrequency(getFrequency());
		target.setNumberOfRepeats(getNumberOfRepeats());
		return target;
	}
```